### PR TITLE
[OTEL-2063] Validate health_check,pprof,zpages in otel flare in E2E test

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?
Validate health_check,pprof,zpages in otel flare in E2E test

### CODE REVIEW

File: test/new-e2e/tests/otel/utils/config_utils.go [hunk 1]
```diff
@@ -42,27 +43,22 @@ func TestOTelFlare(s OTelTestSuite, providedCfg string, fullCfg string, sources
...
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/env.cfg")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/environment.json")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/runtime.cfg")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/runtime_override.cfg")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/health_check/dd-autoconfigured.dat")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_heap.dat")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_allocs.dat")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/command.txt")
+	assert.Contains(s.T(), otelflares, "otel/otel-flare/ext.txt")
+
```
Comment (mackjmr) [inline test/new-e2e/tests/otel/utils/config_utils.go:59]: Looks like you are missing zpages, it should contain:
- otel/otel-flare/zpages/dd-autoconfigured_debug_extensionz.dat
- otel/otel-flare/zpages/dd-autoconfigured_debug_featurez.dat
- otel/otel-flare/zpages/dd-autoconfigured_debug_pipelinez.dat
- otel/otel-flare/zpages/dd-autoconfigured_debug_servicez.dat
- otel/otel-flare/zpages/dd-autoconfigured_debug_tracez.dat

Also you are missing for pprof:
- otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_profile.dat

Comment (songy23) [reply test/new-e2e/tests/otel/utils/config_utils.go:59]: Those are not present in flares in the tests. Does it need any prerequisites? like some specific configs

Comment (songy23) [reply test/new-e2e/tests/otel/utils/config_utils.go:59]: You can see all the otel flares in the test: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/683936374 search `Got otel flare`

Comment (songy23) [reply test/new-e2e/tests/otel/utils/config_utils.go:59]: NVM, I got the wrong link and now I can see them

Comment (songy23) [reply test/new-e2e/tests/otel/utils/config_utils.go:65]: added them all

**Review by mackjmr — APPROVED** [review]

**Review by liustanley — APPROVED** [review]

File: test/new-e2e/tests/otel/utils/config_utils.go [hunk 2]
```diff
@@ -28,52 +30,74 @@ func TestOTelAgentInstalled(s OTelTestSuite) {
...
+	"otel/otel-response.json",
+	"otel/otel-flare/customer.cfg",
+	"otel/otel-flare/env.cfg",
+	"otel/otel-flare/environment.json",
+	"otel/otel-flare/runtime.cfg",
+	"otel/otel-flare/runtime_override.cfg",
+	"otel/otel-flare/health_check/dd-autoconfigured.dat",
+	"otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_heap.dat",
+	"otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_allocs.dat",
+	// "otel/otel-flare/pprof/dd-autoconfigured_debug_pprof_profile.dat",
```
Comment (songy23) [inline test/new-e2e/tests/otel/utils/config_utils.go:43]: zpages have been fixed, cpu profile is the only one remaining. will open a separate tix for it

**Pull request merged**